### ISSUE TITLE
Get content and logistics in place for Firefox hackathon

### ISSUE THREAD

**Issue by MichaelKohler** [mozillach/participation#30]
## Goal:

Have organized everything for the Firefox hackathon in Zurich on April 30th.
## Success:

We are ready to go!
## Related Issues & Links:

[meta] #23 
## Roles:

Owner: @MichaelKohler
Involved: @colinfrei @mte90 @freaktechnik
## Required:
- [x] Define content / format (all, [medium], Due: 15.03.2016) 
- [x] Announce hackathon on meetup and Twitter (@MichaelKohler, medium, Due: 25.03.2016) 
- [x] Promote hackathon (all, medium, Due: 15.04.2016)

**Comment by Mte90** [comment]
For my talks: http://mte90.github.io/
I think that Firefox and WebExtensions and Demolish the web with Firefox are perfect for an hackaton about Firefox.
Maybe https://janitor.technology/ will be useful if we want to hack on Firefox. I tried and works without problems and is very easy. In this way for a developer is not required to have a local system for development/compile.

**Comment by MichaelKohler** [comment]
I filled out a possible agenda here: https://public.etherpad-mozilla.org/p/mozillach-firefox-hackathon-apr16

@Mte90 would you be okay with that?
@freaktechnik @gion-andri do you guys want to have talk as well?

**Comment by Mte90** [comment]
For me is ok the problem of janitor that is in beta, so for use you have to wait for the approval of the account, but for other information we can ask to @jankeromnes.

**Comment by jankeromnes** [comment]
I'd love to give you as much support as possible with https://janitor.technology, but as @Mte90 correctly identified it's still in alpha version (details below).

**The goal**: The Janitor's long-term goal is to support spontaneous contributions to large open source projects like Firefox (e.g. like in the hackathon you're organizing), thanks to unlimited full-featured development environments that spawn in just a few seconds.

**What works**: Currently, I have no problem supporting a full hackathon with on-demand contribution environments (web-accessible high-performance Ubuntu VMs, with the latest Firefox sources pre-compiled and all the tools, and all this can spawn in about 3 seconds), but for now they can only be accessed via _SSH_ for hacking and _noVNC_ for a graphical interface (allows to run the compiled Firefox).

**What doesn't**: What's doesn't work out-of-the-box yet is accessing these containers via the awesome _Cloud9 IDE_, which plays a big part in making the whole experience really awesome. This is because Cloud9 requires you to be a premium user for remote IDE access ("ssh workspaces"), but I'm talking with them to find a solution for the Janitor which doesn't require premiums.

**In summary**, here are our options for your hackathon:
1. Forget about the _Cloud9 IDE_, and just give every participant full Janitor environments accessible via _SSH_ and _noVNC_ only.
2. Talk with Cloud9 to find a way to let Janitor users have the IDE without needing to be premium (we're working on that).
3. a. Ask Cloud9 nicely to grant every participant a few remote workspaces for free, or b. buy everybody a one month premium (both are hard because we need to know everybodys' Cloud9 accounts beforehand).

I'll keep you posted on `2.`: If I fix it in time, I can just give everyone access to https://janitor.technology in April. If not, we'll have to pick a workaround.

**Comment by MichaelKohler** [comment]
http://www.meetup.com/Mozilla-Meetup-Switzerland/events/230065394/
https://public.etherpad-mozilla.org/p/mozillach-firefox-hackathon-apr16

**Comment by MichaelKohler** [comment]
I think we're ready here. Feel free to reopen if there is something else regarding logistics.

**Comment by jankeromnes** [comment]
I think I found a way to let participants use https://janitor.technology for the Hackathon if they're interested in trying out the Alpha.

@MichaelKohler or @Mte90, do you know approximately how many participants there will be? (More like 10, 50, or 200?)

**Comment by MichaelKohler** [comment]
@jankeromnes let's say something between 15 and 20 :) Great news! I really like this idea!

**Comment by Mte90** [comment]
My talk slide about Janitor and dev tools are ready on mte90.github.io :-)

**Comment by jankeromnes** [comment]
Between 15 and 20 sounds perfectly doable! :)

During the event, please mention that anyone interested in trying the Janitor can join the Alpha by:
- Creating a free Cloud9 account at https://c9.io/signup
- Adding an email address to https://janitor.technology
- Joining the [#janitor](https://kiwiirc.com/client/irc.freenode.net/?#janitor) IRC channel on freenode for live support

Thanks!

**Comment by MichaelKohler** [comment]
Thanks @jankeromnes , this will be great! We'll also write a blog post about our experience! :)

**Comment by Mte90** [comment]
ok i update the slides!
